WoWsShipBuilder.ViewModels/Other/ShipComparisonViewModel.cs

```diff
@@ -0,0 +1,762 @@
+            SetSelectAndPinAllButtonsStatus();
+        }
+    }
+
+    public async Task ToggleTierSelection(int value)
+    {
+        if (SelectedTiers.Contains(value))
+        {
+            SelectedTiers.Remove(value);
+        }
+        else
+        {
+            SelectedTiers.Add(value);
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleClassSelection(ShipClass value)
+    {
+        if (SelectedClasses.Contains(value))
+        {
+            SelectedClasses.Remove(value);
+        }
+        else
+        {
+            SelectedClasses.Add(value);
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleNationSelection(Nation value)
+    {
+        if (SelectedNations.Contains(value))
+        {
+            SelectedNations.Remove(value);
+        }
+        else
+        {
+            SelectedNations.Add(value);
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleCategorySelection(ShipCategory value)
+    {
+        if (SelectedCategories.Contains(value))
+        {
+            SelectedCategories.Remove(value);
+        }
+        else
+        {
+            SelectedCategories.Add(value);
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllTiers(bool activationState)
+    {
+        SelectedTiers.Clear();
+        if (activationState)
+        {
+            SelectedTiers.AddRange(Enumerable.Range(1, 11));
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllClasses(bool activationState)
+    {
+        SelectedClasses.Clear();
+        if (activationState)
+        {
+            SelectedClasses.AddRange(Enum.GetValues<ShipClass>().Except(new[] { ShipClass.Auxiliary }));
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllNations(bool activationState)
+    {
+        SelectedNations.Clear();
+        if (activationState)
+        {
+            SelectedNations.AddRange(Enum.GetValues<Nation>().Except(new[] { Nation.Common }));
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllCategories(bool activationState)
+    {
+        SelectedCategories.Clear();
+        if (activationState)
+        {
+            SelectedCategories.AddRange(Enum.GetValues<ShipCategory>().Except(new[] { ShipCategory.Disabled, ShipCategory.Clan }));
+        }
+
+        await ApplyFilters();
+    }
+
+    public List<ShipComparisonDataWrapper> GetShipsToBeDisplayedList(bool disableHideShipsIfNoSelectedSection = false)
+    {
+        List<ShipComparisonDataWrapper> list = ShowPinnedShipsOnly ? PinnedShipList : FilteredShipList;
+
+        if (!disableHideShipsIfNoSelectedSection)
+        {
+            list = HideShipsIfNoSelectedSection(list);
+        }
+
+        return list;
+    }
+
+    public void EditBuilds(List<ShipComparisonDataWrapper> newWrappers, bool clearCache = false)
+    {
+        foreach (var wrapper in newWrappers)
```

Comment:
This logic may be changed entirely and become more efficient by changing the lists as mentioned in a comment at the start of the file